upload-current.sh

```diff
@@ -1,77 +1,89 @@
 #!/usr/bin/env bash
+#
+# Upload trained model s3 in a format compatible with DeepRacer model import functionality.
+# Example usage to upload the best model:
+#     ./upload-current.sh aws-deepracer-XXX model1 -b
+#
 
 S3_BUCKET=$1
 S3_PREFIX=$2
 
 MODEL_DIR=data/minio/bucket/current/model/
 
-while getopts ":c:" opt; do
+echo "Uploading to model ==>  s3://$S3_BUCKET/$S3_PREFIX  <=="
+
+USE_BEST=false
+while getopts ":c:b" opt; do
 case $opt in
-c) CHECKPOINT="$OPTARG"
-;;
-\?) echo "Invalid option -$OPTARG" >&2
-;;
+    c) CHECKPOINT="$OPTARG"
+       ;;
+    b) USE_BEST=true
+       ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+        ;;
 esac
 done
 
 CHECKPOINT_FILE=$MODEL_DIR"deepracer_checkpoints.json"
 if [ ! -f ${CHECKPOINT_FILE} ]; then
   echo "Checkpoint file not found!"
   exit 1
-else
-  echo "found checkpoint index file "$CHECKPOINT_FILE
-fi;
+fi
+echo "found checkpoint index file "$CHECKPOINT_FILE
 
 if [ -z "$CHECKPOINT" ]; then
-  echo "Checkpoint not supplied, checking for latest checkpoint"
-
-  LAST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".last_checkpoint.name"`
-  BEST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".best_checkpoint.name"`
-  
-  CHECKPOINT=$LAST_CHECKPOINT
-
-  echo "latest checkpoint = "$CHECKPOINT
+  #echo "Checkpoint not supplied, checking for latest checkpoint"
+  LAST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".last_checkpoint.name" | sed s/\"//g`
+  BEST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".best_checkpoint.name" | sed s/\"//g`
+  if $USE_BEST; then
+     CHECKPOINT=$BEST_CHECKPOINT
+     echo "Using best checkpoint ==>  $CHECKPOINT  <=="
+  else
+     CHECKPOINT=$LAST_CHECKPOINT
+     echo "Using latest checkpoint ==>  $CHECKPOINT  <=="
+  fi
 else
   echo "Checkpoint supplied: ["${CHECKPOINT}"]"
 fi
 
 MODEL=`echo $CHECKPOINT |sed "s@^[^0-9]*\([0-9]\+\).*@\1@"`
-mkdir -p checkpoint
+rm -rf checkpoint
+cp -a upload-template checkpoint
+mkdir -p checkpoint/model
 MODEL_FILE=$MODEL_DIR"model_"$MODEL".pb"
 METADATA_FILE=$MODEL_DIR"model_metadata.json"
 
-
 if test ! -f "$MODEL_FILE"; then
     echo "$MODEL_FILE doesn't exist"
     exit 1
-else
-  cp $MODEL_FILE checkpoint/  
 fi
 
 if test ! -f "$METADATA_FILE"; then
     echo "$METADATA_FILE doesn't exist"
     exit 1
-else
-  cp $METADATA_FILE checkpoint/  
 fi
 
-CHECKPOINT_FILES=`echo $CHECKPOINT* |sed "s/\"//g"`
-for i in $( find $MODEL_DIR -type f -name $CHECKPOINT_FILES ); do
-  cp $i checkpoint/  
+cp -v $MODEL_FILE checkpoint/model/
+cp -v $METADATA_FILE checkpoint/model/
+
+CHECKPOINT_FILES=$MODEL_DIR/${CHECKPOINT}*
```

Comment:
It seems tha according to the spec we need current checkpoint and previous one (so 2 checkpoint sets) - so the previous checkpoint should be copied or simply all checkpoints (there are two sets per last and best anyways), i.e. `CHECKPOINT_FILES=$MODEL_DIR/*.ckpt.*`

Reply:
It works for race submission with a single checkpoint. Does something break without 2 checkpoints?

If we really need multiple checkpoints then we should upload all checkpoints.